src/databricks/labs/ucx/hive_metastore/migration_status.py

```diff
@@ -0,0 +1,129 @@
+import datetime
+import logging
+from dataclasses import dataclass
+from collections.abc import Iterable
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MigrationStatus:
+    src_schema: str
+    src_table: str
+    dst_catalog: str | None = None
+    dst_schema: str | None = None
+    dst_table: str | None = None
+    update_ts: str | None = None
+
+    def destination(self):
```

Comment:
`@property` probably makes sense here, used within other data classes as well

Reply:
Why do we need property?